src/engine/SpatialJoin.cpp

```diff
@@ -349,8 +363,9 @@ PreparedSpatialJoinParams SpatialJoin::prepareJoin() const {
     return std::pair{idTablePtr, std::move(resTable)};
   };
 
-  auto getJoinCol = [](VariableToColumnMap varColMap,
+  auto getJoinCol = [](std::shared_ptr<QueryExecutionTree> child,
                        const Variable& childVariable) {
+    VariableToColumnMap varColMap = child->getVariableColumns();
     return varColMap[childVariable].columnIndex_;
```

Comment:
```suggestion
     child->getVariableColumn(childVariable);
```
Then you can easily complete delete the lambda, because the code is so short...

You rcurrent code still has an unnecessary copy of the VarToColMap (at least use `const auto&`
And the problem with case 2 above. (if you don't use my first solution, at least use the `at ` fucntion instead of `operator[]`.

Reply:
My bad, I confused two locations in the code. This is now fixed like you suggested.